src/unit.rs

```diff
@@ -378,42 +377,117 @@ fn send_prelude(unit: &Unit, stream: &mut Stream, redir: bool) -> io::Result<()>
                     _ => 0,
                 };
                 if scheme_default != 0 && scheme_default == port {
-                    write!(prelude, "Host: {}\r\n", host)?;
+                    prelude.write_header("Host", host)?;
                 } else {
-                    write!(prelude, "Host: {}:{}\r\n", host, port)?;
+                    prelude.write_header("Host", format_args!("{}:{}", host, port))?;
                 }
             }
             None => {
-                write!(prelude, "Host: {}\r\n", host)?;
+                prelude.write_header("Host", host)?;
             }
         }
     }
     if !header::has_header(&unit.headers, "user-agent") {
-        write!(prelude, "User-Agent: {}\r\n", &unit.agent.config.user_agent)?;
+        prelude.write_header("User-Agent", &unit.agent.config.user_agent)?;
     }
     if !header::has_header(&unit.headers, "accept") {
-        write!(prelude, "Accept: */*\r\n")?;
+        prelude.write_header("Accept", "*/*")?;
     }
 
     // other headers
     for header in &unit.headers {
         if !redir || !header.is_name("Authorization") {
             if let Some(v) = header.value() {
-                write!(prelude, "{}: {}\r\n", header.name(), v)?;
+                if is_header_sensitive(header) {
+                    prelude.write_sensitive_header(header.name(), v)?;
+                } else {
+                    prelude.write_header(header.name(), v)?;
+                }
             }
         }
     }
 
     // finish
-    write!(prelude, "\r\n")?;
+    prelude.finish()?;
 
-    debug!("writing prelude: {}", String::from_utf8_lossy(&prelude));
+    debug!("writing prelude: {:?}", prelude);
```

Comment:
Hm hm… I think we maybe should use the `Display` trait here and let the `Debug` trait just dump the internal state of the `PreludeBuilder`.

Reply:
Changed to use `Display` instead